test/integration/controller/core/workload_controller_test.go

```diff
@@ -142,4 +140,141 @@ var _ = ginkgo.Describe("Workload controller", func() {
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, clusterQueue)).To(gomega.Succeed())
+		})
+
+		ginkgo.It("Should accumulate RuntimeClass's overhead", func() {
+			ginkgo.By("Create workload")
+			wl = testing.MakeWorkload("one", ns.Name).
+				Queue(queue.Name).
+				Request(corev1.ResourceCPU, "1").
+				Admit(testing.MakeAdmission(clusterQueue.Name).
+					Flavor(corev1.ResourceCPU, flavorOnDemand).Obj()).
+				RuntimeClass("kata").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+
+			ginkgo.By("Got ClusterQueueStatus")
+			gomega.Eventually(func() kueue.ClusterQueueStatus {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterQueue), &updatedCQ)).To(gomega.Succeed())
+				return updatedCQ.Status
+			}, framework.Timeout, framework.Interval).Should(testing.Equal(kueue.ClusterQueueStatus{
+				PendingWorkloads:  0,
+				AdmittedWorkloads: 1,
+				UsedResources: kueue.UsedResources{
+					corev1.ResourceCPU: {
+						flavorOnDemand: {
+							Total:    pointer.Quantity(resource.MustParse("2")),
+							Borrowed: nil,
+						},
+					},
+				},
+			}))
+		})
+	})
+
+	ginkgo.When("Workload with non-exists RuntimeClass defined", func() {
+		ginkgo.BeforeEach(func() {
+			clusterQueue = testing.MakeClusterQueue("clusterqueue").
+				Resource(testing.MakeResource(corev1.ResourceCPU).
+					Flavor(testing.MakeFlavor(flavorOnDemand, "5").Max("10").Obj()).Obj()).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+			queue = testing.MakeQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, queue)).To(gomega.Succeed())
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, clusterQueue)).To(gomega.Succeed())
+		})
+
+		ginkgo.It("Should not accumulate RuntimeClass's overhead", func() {
+			ginkgo.By("Create workload")
+			wl = testing.MakeWorkload("one", ns.Name).
+				Queue(queue.Name).
+				Request(corev1.ResourceCPU, "1").
+				Admit(testing.MakeAdmission(clusterQueue.Name).
+					Flavor(corev1.ResourceCPU, flavorOnDemand).Obj()).
+				RuntimeClass("kata").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+
+			ginkgo.By("Got ClusterQueueStatus")
+			gomega.Eventually(func() kueue.ClusterQueueStatus {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterQueue), &updatedCQ)).To(gomega.Succeed())
+				return updatedCQ.Status
+			}, framework.Timeout, framework.Interval).Should(testing.Equal(kueue.ClusterQueueStatus{
+				PendingWorkloads:  0,
+				AdmittedWorkloads: 1,
+				UsedResources: kueue.UsedResources{
+					corev1.ResourceCPU: {
+						flavorOnDemand: {
+							Total:    pointer.Quantity(resource.MustParse("1")),
+							Borrowed: nil,
+						},
+					},
+				},
+			}))
+		})
+	})
+
+	ginkgo.When("workload with PodOverhead defined, RuntimeClass undefined", func() {
```

Comment:
I would just remove this test.

Reply:
That's why we should remove it :)

It's just non-useful code we shouldn't have to maintain.